torchgeo/transforms/intensity.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""TorchGeo intensity transforms."""
+
+import torch
+from kornia.augmentation import IntensityAugmentationBase2D
+from torch import Tensor
+
+
+class ToDecibelScale(IntensityAugmentationBase2D):
+    """Convert input tensor of power values to decibel scale.
+
+    Primarily used for converting SAR pixel values to decibel scale.
+
+    .. versionadded:: 0.8
+    """
+
+    def __init__(
+        self, shift: float = 0.0, scale: float = 1.0, keepdim: bool = False
+    ) -> None:
+        """Initialize a new ToDecibelScale instance.
+
+        Args:
+            shift: The value to add to the decibel scale.
+                This is useful for avoiding negative values.
+            scale: An additional scale factor to apply to the decibel value.
+            keepdim: Whether to keep the output shape the same as input (True)
+                or broadcast it to the batch form (False).
+        """
+        super().__init__(p=1.0, p_batch=1.0, same_on_batch=True, keepdim=keepdim)
+        self.flags = {'scale': scale, 'shift': shift}
+
+    def apply_transform(
+        self,
+        input: Tensor,
+        params: dict[str, Tensor],
+        flags: dict[str, Tensor],
+        transform: Tensor | None = None,
+    ) -> Tensor:
+        """Apply the transform.
+
+        Args:
+            input: The input tensor.
+            params: Generated parameters.
+            flags: Static parameters.
+            transform: The geometric transformation tensor.
+
+        Returns:
+            The transformed input.
+        """
+        out = flags['scale'] * 10 * torch.log10(input + 1e-6) + flags['shift']
+        return out
+
+
+class Sentinel1ChangeMap(IntensityAugmentationBase2D):
```

Comment:
If this is general purpose, maybe we can rename it to `Threshold` or something like that? Since it doesn't have to be S1 or CD.

Reply:
Yeah I think I can likely make it more general like take in a list of thresholds so it's not fixed to 4 bands

Reply:
It's not only thresholding though. It's generating a binary change mask by taking the first half of the channels and subtracting them from the second half of the channels.

Reply:
Maybe 2 transforms? One for change detection/comparison and one for thresholding? Idk, don't want to make this too complicated. Use your best judgement.

Reply:
Could also move the subtraction part back into the model and only have a post-processing transform for thresholding if that makes more sense. Really just depends on how general purpose this is. I don't want transforms that only make sense and will only be used with one model.

I assume this part of the model has no learnable weights, so it doesn't matter whether this goes in the model or in a transform?